packages/components/src/tooltip/test/index.tsx

```diff
@@ -312,6 +304,11 @@ describe( 'Tooltip', () => {
 			// Hover outside of the anchor, tooltip should hide
 			await hoverOutside();
 			await waitExpectTooltipToHide();
+
+			// Prevent this test from interfering with the next one.
+			// "Tooltips appear instantly if another tooltip has just been hidden."
+			// See: https://github.com/WordPress/gutenberg/pull/57345#discussion_r1435495655
+			await sleep( 3000 );
```

Comment:
Ideally, there would be a better way, and something more specific to wait for, but this is not related to the PR, so 🤷

Reply:
I believe this won't be needed with `@ariakit/react@^0.4.2`.

Reply:
Oh I didn't realize we're still on 0.3.12, is there any discussion on upgrading? Any relevant blockers? @tyxla

Reply:
I think there were some `v0.4.0` breaking changes we need to address, like:

- We may have to manually set `role="gridcell"` to grid cells within `Composite` when  using `role="grid"`
- Data attributes can no longer be empty strings
- Type changes

Nothing too critical though - I'd go through all the breaking changes and if there are any that apply to the package, verify that we update properly without breaking something. 

Also I haven't been directly involved with updating Ariakit before, so perhaps @mirka might know more.

Reply:
For the protocol, I'd love for us to update to the latest version. There are so many improvements and new features we could benefit from!